File: eval/rule_accuracy/corpus/go/CWE-89/tp_sql_injection.go
// SQL injection patterns — true positives for rule accuracy testing
package corpus

import "database/sql"

func badSQLConcat(db *sql.DB, userID string) {
	db.Query("SELECT * FROM users WHERE id = '" + userID + "'") // expect: go-sql-string-concat
}

func badSQLSprintf(db *sql.DB, name string) {
	query := fmt.Sprintf("SELECT * FROM users WHERE name = '%s'", name)
	db.Query(query) // expect: go-sql-string-concat
}

File: eval/rule_accuracy/corpus/java/CWE-89/tp_sql_injection.java
// SQL injection — true positives for rule accuracy testing
// INTENTIONALLY insecure — nosec

public class SqlInjection {
    public void badCreateQuery(String userInput) {
        em.createQuery("SELECT u FROM User u WHERE u.name = '" + userInput + "'");  // expect: java/java-sql-concat
    }

    public void badStatement(String id) throws Exception {
        Statement stmt = conn.createStatement();
        stmt.executeQuery("SELECT * FROM users WHERE id = " + id);  // expect: security/java_sql_concat
    }
}
